src/jit/codegenarm64.cpp

```diff
@@ -4752,7 +4752,7 @@ void CodeGen::genLeaInstruction(GenTreeAddrMode* lea)
                 instGen_Set_Reg_To_Imm(EA_PTRSIZE, tmpReg, offset);
                 // Then add the base register
                 //      rd = rd + base
-                emit->emitIns_R_R_R(INS_add, EA_PTRSIZE, tmpReg, tmpReg, memBase->gtRegNum);
+                emit->emitIns_R_R_R(INS_add, size, tmpReg, tmpReg, memBase->gtRegNum);
```

Comment:
In all three cases above , size is the same as the ultimate emitAttr for the LeaInstruction.  As soon as the Base() register is added it acquires its final GC attr.  

Longer term it would be better to fix the CQ issue cited at the top of this function and not generate these extra instructions.